### ISSUE TITLE
Crashes after selecting date range

### ISSUE THREAD

**Issue by J-Sek** [nlogozzo/NickvisionMoney#249]
**Reproduction**:
1. Open account
2. Click on Select Range toggle below the calendar

> Let me know where I can find the logs or how to start the app from the terminal to see the exception

**Current Result**: Application crash (every time)

**When did it started?**
After recent update (rewrite from Money to Denaro).
- yes, I used the previous account file

**System information**
PC, Zorin OS (Ubuntu 20.04)

**Comment by fsobolev** [comment]
> Let me know where I can find the logs or how to start the app from the terminal to see the exception

`flatpak run org.nickvision.money//stable`

**Comment by nlogozzo** [comment]
Also, please provide the version of Denaro that you are running and make sure you are running the latest stable.

*Remember if you installed Denaro from `flathub-beta` you will not get the latest stable version. You will need to install `Denaro` from stable `flathub` repo*

**Comment by J-Sek** [comment]
Hi. Thanks for the command.
My about page is showing `2023.1.1`. It is a stable release.

```log
Registering new type NickvisionMoneyGNOMEViewsMainWindow with parent AdwApplicationWindow
Registering new type NickvisionMoneyGNOMEControlsGroupRow with parent AdwActionRow
Unhandled exception. System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at NickvisionMoney.GNOME.Views.AccountView.OnDateRangeStartYearChanged() in /run/build/org.nickvision.money/NickvisionMoney.GNOME/Views/AccountView.cs:line 1260
   at NickvisionMoney.GNOME.Views.AccountView.<.ctor>b__75_4(Object sender, NotifySignalArgs e) in /run/build/org.nickvision.money/NickvisionMoney.GNOME/Views/AccountView.cs:line 296
   at GObject.Signal`2.<>c__DisplayClass7_0.<Connect>b__0(Value returnValue, Value[] parameters)
```

**Comment by fsobolev** [comment]
I managed to reproduce the bug with absolutely new, clean account without transactions, the error is exactly the same.

**Comment by nlogozzo** [comment]
@J-Sek did the account that you opened also have 0 transactions?

**Comment by nlogozzo** [comment]
> I managed to reproduce the bug with absolutely new, clean account without transactions, the error is exactly the same.

Fixed with https://github.com/nlogozzo/NickvisionMoney/commit/d63d4b850145a08a88afcaf5926ccabe8a732076

**Comment by J-Sek** [comment]
> @J-Sek did the account that you opened also have 0 transactions?

I guess I had a few in 2022, so they did not appear for current month... but I may be wrong.
I don't input transactions because I evaluate a way to import them instead of keeping track manually.
Thanks for the fix :)

**Comment by nlogozzo** [comment]
> Thanks for the fix :)

No problem. It's not fixed in 2023.2.0-beta1, there is a little bug...but it's fixed in main and will be available in beta2 soon :). I'll send a link here when it's released!

**Comment by nlogozzo** [comment]
https://github.com/nlogozzo/NickvisionMoney/releases/tag/2023.2.0-beta2
